File: app/src/main/java/com/mapmyindia/sdk/demo/java/activity/DirectionPluginActivity.java
package com.mapmyindia.sdk.demo.java.activity;

import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import com.mapmyindia.sdk.demo.R;
import com.mapmyindia.sdk.demo.databinding.ActivityDirectionPluginBinding;
import com.mapmyindia.sdk.direction.ui.DirectionCallback;
import com.mapmyindia.sdk.direction.ui.DirectionFragment;
import com.mapmyindia.sdk.direction.ui.model.DirectionOptions;
import com.mapmyindia.sdk.direction.ui.model.DirectionPoint;
import com.mmi.services.api.directions.models.DirectionsResponse;

import java.util.List;

public class DirectionPluginActivity extends AppCompatActivity {

    private ActivityDirectionPluginBinding mBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_direction_plugin);

        mBinding.btnOpenDirectionFragmentUi.setOnClickListener(v ->{


            DirectionOptions options = DirectionOptions.builder().build();
            DirectionFragment directionFragment = DirectionFragment.newInstance(options);


            getSupportFragmentManager().
                    beginTransaction().
                    add(R.id.container, directionFragment, DirectionFragment.class.getSimpleName()).
                    addToBackStack(null).
                    commit();

            directionFragment.setDirectionCallback(new DirectionCallback() {
                @Override
                public void onCancel() {
                     getSupportFragmentManager().beginTransaction().remove(directionFragment).commit();
                }

                @Override
                public void onStartNavigation(DirectionPoint directionPoint, DirectionPoint directionPoint1, List<DirectionPoint> list, DirectionsResponse directionsResponse, int i) {
                    Toast.makeText(DirectionPluginActivity.this, "On Navigation Start", Toast.LENGTH_SHORT).show();
                }
            });
        });


    }
}

File: app/src/main/java/com/mapmyindia/sdk/demo/kotlin/activity/DirectionPluginActivity.kt
package com.mapmyindia.sdk.demo.kotlin.activity

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.mapmyindia.sdk.demo.R
import com.mapmyindia.sdk.demo.databinding.ActivityDirectionPluginBinding
import com.mapmyindia.sdk.direction.ui.DirectionCallback
import com.mapmyindia.sdk.direction.ui.DirectionFragment
import com.mapmyindia.sdk.direction.ui.model.DirectionOptions
import com.mapmyindia.sdk.direction.ui.model.DirectionPoint
import com.mmi.services.api.directions.models.DirectionsResponse


class DirectionPluginActivity : AppCompatActivity() {
    private lateinit var mBinding: ActivityDirectionPluginBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_direction_plugin)

        mBinding.btnOpenDirectionFragmentUi.setOnClickListener { v ->


            val options = DirectionOptions.builder().build()

            val directionFragment = DirectionFragment.newInstance(options)
            supportFragmentManager.
            beginTransaction().
            add(R.id.container, directionFragment, DirectionFragment::class.java.simpleName).
            addToBackStack(null).
            commit()

            directionFragment.setDirectionCallback(object : DirectionCallback {
                override fun onCancel() {
                    supportFragmentManager.beginTransaction().remove(directionFragment).commit()
                }

                override fun onStartNavigation(directionPoint: DirectionPoint, directionPoint1: DirectionPoint, list: List<DirectionPoint>, directionsResponse: DirectionsResponse, i: Int) {
                    Toast.makeText(this@DirectionPluginActivity, "On Navigation Start", Toast.LENGTH_SHORT).show()
                }
            })
        }
    }
}